src/pi-gen-config.ts

```diff
@@ -92,16 +92,17 @@ export async function validateConfig(config: PiGenConfig): Promise<void> {
     )
   }
 
-  if (
-    !['none', 'zip', 'gz', 'xz'].includes(
-      config.deployCompression?.toLowerCase()
-    )
-  ) {
+  const deployCompression = config.deployCompression?.toLowerCase()
+  if (!['none', 'zip', 'gz', 'xz'].includes(deployCompression)) {
     throw new Error('compression must be one of ["none", "zip", "gz", "xz"]')
   }
 
-  if (!/^[0-9]$/.test(config.compressionLevel)) {
-    throw new Error('compression-level must be between 0 and 9')
+  if (
+    !/^[0-9]$/.test(config.compressionLevel) &&
```

Comment:
Shouldn't these two conditions be joined using `or`? 
```suggestion
    !/^[0-9]$/.test(config.compressionLevel) ||
```

Reply:
Pretty sure this is correct

Throw an error when

* (`compressionLevel` does NOT match `^[0-9]$`) and (`deployCompression` is `"xz"` and `compressionLevel` is not `"9e"`)

Reply:
Woops. I think you're right. Let me take a second look.